universal-login-commons/src/integration/ethereum/ProviderService.ts

```diff
@@ -4,11 +4,19 @@ import {fetchHardforkVersion} from './fetchHardforkVersion';
 import {NetworkVersion} from '../../core/utils/messages/computeGasData';
 
 export class ProviderService {
+  private cachedContractCodes: Map<string, Promise<string>>;
```

Comment:
You can make it in a similar way as there: WalletContractService

Reply:
Done